clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2185,9 +2185,157 @@ void handleFailure(Throwable throwable) {
                 completeAllExceptionally(topicFutures.values(), throwable);
             }
         };
-        if (!topicNamesList.isEmpty()) {
-            runnable.call(call, now);
+    }
+
+    Call generateDescribeTopicsCallWithDescribeTopicPartitionsApi(
+        List<String> topicNamesList,
+        Map<String, KafkaFutureImpl<TopicDescription>> topicFutures,
+        Map<Integer, Node> nodes,
+        DescribeTopicsOptions options,
+        long now
+    ) {
+        Map<String, TopicRequest> topicsRequests = new LinkedHashMap<>();
+        topicNamesList.stream().sorted().forEach(topic -> {
+            topicsRequests.put(topic, new TopicRequest().setName(topic));
+        });
+        return new Call("describeTopicPartitions", calcDeadlineMs(now, options.timeoutMs()),
+            new LeastLoadedNodeProvider()) {
+            Map<String, TopicRequest> pendingTopics = topicsRequests;
```

Comment:
Why do we need this inner assignment? Can we make `topicsRequest` final in the outer scope?